httpcache.go

```diff
@@ -100,15 +120,21 @@ type Transport struct {
 	// The RoundTripper interface actually used to make requests
 	// If nil, http.DefaultTransport is used
 	Transport http.RoundTripper
-	Cache     Cache
+	Cache     StreamingCache
```

Comment:
Well, we could leave this as `Cache` and then:

```go
cache, ok := tp.Cache.(StreamingCache)
if !ok {
  cache = NewBufferedStreamingCache(tp.Cache)
}
```

before consuming it.  Or we could create a new `StreamingCacheTransport` and make the `Transport` methods backwards-compat wrappers around the `StreamingCacheTransport` methods.

Did you have a deprecation timescale in mind?  Dropping `NewBufferedStreamingCache` wrappers around existing `Cache` implementations seems like it would be easy for consumers to do.

Reply:
The first option seems reasonable.

I don't think there will be any deprecation, I consider the API to be set as it is, and having people have to make changes, even small ones, goes against that contract. A new API would have to come with a new import path.